### DISCUSSION TITLE
Why isn't my RLS allowing the user to update a row?

### DISCUSSION THREAD

**Discussion by gurkang** [supabase/supabase#13391] [Questions]
I'm trying to figure out the RLS thing in supabase and it's not working like I'm expecting.

This is my policy:
![Screenshot 2023-04-03 at 19 41 12](https://user-images.githubusercontent.com/9166287/229512412-f95e459e-61b9-4ee7-9753-4937e551f326.png)

And my table:

```
id | No description | bigint | int8 |  
created_at | No description | timestamp with time zone | timestamptz |  
downloads_left | No description | bigint | int8 |  
userId | No description | uuid | uuid |  
```

And the query I'm trying to run:

```
await supabaseServerClient
                .from("extendedUser")
                .update({
                  downloads_left:
                    0,
                })
                .eq("userId", id);
```

When I disable RLS on the table it works just fine. But when I enable RLS with the above policy it just won't update.
It doesn't throw any errors and the return data from the query is just a `204`.

I'm really stumped here cause I though it would work by checking that the` auth.uid ` equals my tables `userId`.

**Comment by GaryAustin1** [comment]
Is that policy for all operations ? You need to meet both update and select policy. 

You seem to be calling from a server.  How is your user jwt getting setup into the server client?  You may need to show createClient depending on if you are using auth-helpers or not.  You should verify you have a user session when you make the call.  You could always change your RLS to Roles as anon and true for the policy as a test.  If that works you are not signed in.

**Reply by gurkang** [reply]
I start by creating a supabase server client like so inside my nextjs api folder:
```
  const supabaseServerClient = createServerSupabaseClient<Database>({
      req,
      res,
    });
```


I then set the session by sending along the refreshToken and acessToken:
```
const user = await supabaseServerClient.auth.setSession({
        refresh_token: refresh_token,
        access_token: token,
      });
```

When I inspect this user object (before making the requests) it says user is authenticated and is a valid session. So I should be authenticated and it should be working right?

Or am I missing something here.

**Comment by GaryAustin1** [comment]
I don't know much about auth-helpers, but that seems like what you are using

![image](https://user-images.githubusercontent.com/54564956/229521500-7a874764-5ce6-43ef-a600-ed20249a274b.png)

I'm not sure why you would use setSession if you are using helpers.  The pictured code shows one example and it uses getSession.

Also doing the RLS test I suggested will confirm if you are logged in or not when the call is made.
